### PR TITLE
[Serve] Add cosmetic improvements to `serve build`

### PR DESCRIPTION
**Pull request by shrekris-anyscale**
## Why are these changes needed?

`serve build` is a Serve CLI command that generates a Serve config from a deployment graph node. This change adds some cosmetic improvements to its output:

* Adds spacing throughout the config file
* Appends a comment to the file, which specifies the Ray version used by `serve build`

The new format is:

```yaml
# This file was generated using the `serve build` command on Ray v3.0.0.dev0.

import_path: example.path

runtime_env: {}

deployments:

- name: val1
  num_replicas: 1
  ...

- name: val2
  num_replicas: 1
  ...

```

## Related issue number

N/A

## Checks

- [X] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - This change relies on existing unit tests and adds only cosmetic improvements.

### CODE REVIEW

File: python/ray/serve/scripts.py [hunk 1]
```diff
@@ -397,11 +403,40 @@ def build(import_path: str, app_dir: str, output_path: Optional[str]):
             if property in deployment:
                 del deployment[property]
 
-    if output_path is not None:
-        if not output_path.endswith(".yaml"):
-            raise ValueError("FILE_PATH must end with '.yaml'.")
+    config_str = yaml.dump(
+        config, Dumper=ServeBuildDumper, default_flow_style=False, sort_keys=False
+    )
+    config_str += f"# Built using Ray v{ray.__version__}\n"
```
Comment (simon-mo) [inline python/ray/serve/scripts.py:409]: We should put this at the top of the file. And add a line saying "This file is generated using `serve build` command" or something like that.

File: python/ray/serve/scripts.py [hunk 2]
```diff
@@ -397,11 +403,40 @@ def build(import_path: str, app_dir: str, output_path: Optional[str]):
...
 
+    if output_path is not None:
         with open(output_path, "w") as f:
-            yaml.safe_dump(config, stream=f, default_flow_style=False, sort_keys=False)
+            f.write(config_str)
     else:
-        print(yaml.safe_dump(config, default_flow_style=False, sort_keys=False), end="")
+        print(config_str, end="")
```
Comment (simon-mo) [inline python/ray/serve/scripts.py:415]: you can unify this with default f to sys.stdout

File: python/ray/serve/scripts.py [hunk 3]
```diff
@@ -397,11 +403,40 @@ def build(import_path: str, app_dir: str, output_path: Optional[str]):
...
+    - name: val2
+        ...
+    """
+
+    def write_line_break(self, data=None):
+        # https://github.com/yaml/pyyaml/issues/127#issuecomment-525800484
+        super().write_line_break(data)
+
+        if len(self.indents) < 3:
+            super().write_line_break()
```
Comment (simon-mo) [inline python/ray/serve/scripts.py:442]: what does this do?

**Review by simon-mo — APPROVED** [review]

File: python/ray/serve/scripts.py (hunk 3 above)
Comment (shrekris-anyscale) [reply python/ray/serve/scripts.py:442]: It adds line breaks between the deployment configurations and the top-level fields. It makes the YAML file look like this:

```yaml
import_path: ...

runtime_env: ...

deployments:

- name: deployment1
  num_replicas: 1
  max_concurrent_queries: 10
  ...

- name: deployment2
  num_replicas: 2
  max_concurrent_queries: 3
  ...
```

Without it, the file looks like this:

```yaml
import_path: ...
runtime_env: ...
deployments:
- name: deployment1
  num_replicas: 1
  max_concurrent_queries: 10
  ...
- name: deployment2
  num_replicas: 2
  max_concurrent_queries: 3
  ...
```

File: python/ray/serve/scripts.py (hunk 1 above)
Comment (shrekris-anyscale) [reply python/ray/serve/scripts.py]: I moved the comment to the top of the file and changed it to say:

```yaml
# This file was generated using the `serve build` command on Ray v3.0.0.dev0.
```

File: python/ray/serve/scripts.py (hunk 2 above)
Comment (shrekris-anyscale) [reply python/ray/serve/scripts.py]: Got it, I refactored it to:

```python
with open(output_path, "w") if output_path else sys.stdout as f:
    f.write(config_str)
```

File: python/ray/serve/scripts.py (hunk 3 above)
Comment (shrekris-anyscale) [reply python/ray/serve/scripts.py:437]: I added a comment:

```Python
# Indents must be less than 3 to ensure that only the top 2 levels of
# the config file have line breaks between them. The top 2 levels include
# import_path, runtime_env, deployments, and all entries of deployments.
```

File: python/ray/serve/scripts.py [hunk 4]
```diff
@@ -373,6 +373,12 @@ def shutdown(address: str, yes: bool):
     ),
 )
 def build(import_path: str, app_dir: str, output_path: Optional[str]):
+
+    if output_path is not None and not output_path.endswith(".yaml"):
+        raise ValueError(
+            f'FILE_PATH must end with ".yaml". Got "{output_path}" instead.'
+        )
```
Comment (simon-mo) [inline python/ray/serve/scripts.py:380]: this is too strict. we should not force a suffix.

**Review by simon-mo — APPROVED** [review]

File: python/ray/serve/scripts.py (hunk 4 above)
Comment (shrekris-anyscale) [reply python/ray/serve/scripts.py]: Sounds good, I removed the check.

**Pull request merged**